contracts/UserWithdrawalManager.sol

```diff
@@ -86,12 +86,31 @@ contract UserWithdrawalManager is
         emit UpdatedStaderConfig(_staderConfig);
     }
 
+    /**
+     * @notice auxiliary method to put a withdrawal request
+     * @param _ethXAmount amount of ethX shares to withdraw
+     * @param _owner owner of withdraw request to redeem
+     * @param _referralId referral id of KOL
+     * @return _requestId
+     * @dev emits an event with referralId
+     */
+    function requestWithdrawViaKOL(
+        uint256 _ethXAmount,
+        address _owner,
+        string calldata _referralId
+    ) external override whenNotPaused returns (uint256 _requestId) {
+        uint256 etherAmount = IStaderStakePoolManager(staderConfig.getStakePoolManager()).previewWithdraw(_ethXAmount);
```

Comment:
we can remove this as this amount is also emitting inside requestWithdraw event

Reply:
It would make things complicated for KOLs to analyse, as then they would have to track two events/methods to extract complete info.
But yes I agree, it increases the gas cost.

Reply:
How about returning ethAmount as well from `requestWithdraw` ?

Reply:
make sense, we can keep it then

Reply:
yeah it will break our integration flow